### PR TITLE
vendor: Bump vishvananda/netlink dependency

### PR DESCRIPTION
**Pull request by tgraf**
Neighbor insertion can block indefinitely. See
https://github.com/cilium/cilium/issues/14710 for details.

```
goroutine 62 [syscall, 1080 minutes]:
syscall.Syscall6(0x2d, 0x4e, 0xc002e27000, 0x1000, 0x0, 0xc004ba33e0, 0xc004ba33d4, 0x20, 0x21b1380, 0xc005241101)
        /usr/local/go/src/syscall/asm_linux_amd64.s:44 +0x5
syscall.recvfrom(0x4e, 0xc002e27000, 0x1000, 0x1000, 0x0, 0xc004ba33e0, 0xc004ba33d4, 0x27637a0, 0xc0015555c0, 0x0)
        /usr/local/go/src/syscall/zsyscall_linux_amd64.go:1618 +0xa3
syscall.Recvfrom(0x4e, 0xc002e27000, 0x1000, 0x1000, 0x0, 0x4, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/syscall/syscall_unix.go:273 +0xaf
syscall.NetlinkRIB(0x12, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/syscall/netlink_linux.go:71 +0x23c
net.interfaceTable(0x0, 0x4, 0x4, 0x8, 0x18, 0xc002462cc0)
        /usr/local/go/src/net/interface_linux.go:17 +0x48
net.InterfaceByName(0xc0001dd298, 0x4, 0xc00a3e6a20, 0x10, 0x10)
        /usr/local/go/src/net/interface.go:157 +0x1a4
github.com/cilium/cilium/pkg/datapath/linux.(*linuxNodeHandler).insertNeighbor(0xc0008d4c30, 0xc004ba3b58, 0xc0001dd298, 0x4)
        /go/src/github.com/cilium/cilium/pkg/datapath/linux/node.go:591 +0x132
github.com/cilium/cilium/pkg/datapath/linux.(*linuxNodeHandler).nodeUpdate(0xc0008d4c30, 0x0, 0xc004ba3b58, 0x0, 0x0, 0x0)
        /go/src/github.com/cilium/cilium/pkg/datapath/linux/node.go:729 +0x18f
github.com/cilium/cilium/pkg/datapath/linux.(*linuxNodeHandler).NodeValidateImplementation(0xc0008d4c30, 0xc0006a6ed0, 0x29, 0xc00a3e6958, 0x7, 0xc00be311d0, 0x2, 0x2, 0xc001646368, 0x0, ...)
        /go/src/github.com/cilium/cilium/pkg/datapath/linux/node.go:1171 +0x9b
github.com/cilium/cilium/pkg/node/manager.(*Manager).backgroundSync.func1(0x27c8680, 0xc0008d4c30)
        /go/src/github.com/cilium/cilium/pkg/node/manager/manager.go:282 +0xaa
github.com/cilium/cilium/pkg/node/manager.(*Manager).Iter(0xc00091c280, 0xc004ba3e08)
        /go/src/github.com/cilium/cilium/pkg/node/manager/manager.go:149 +0xe7
github.com/cilium/cilium/pkg/node/manager.(*Manager).backgroundSync(0xc00091c280)
        /go/src/github.com/cilium/cilium/pkg/node/manager/manager.go:281 +0x421
created by github.com/cilium/cilium/pkg/node/manager.NewManager
        /go/src/github.com/cilium/cilium/pkg/node/manager/manager.go:191 +0x673
```

### CODE REVIEW

**Comment by tgraf** [comment]
test-backport-1.7

File: pkg/datapath/linux/route/route_linux.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func (r *Route) getNetlinkRoute() netlink.Route {
 		Dst:      &r.Prefix,
 		Src:      r.Local,
 		MTU:      r.MTU,
-		Protocol: r.Proto,
+		Protocol: netlink.RouteProtocol(r.Proto),
```
Comment (joestringer) [inline pkg/datapath/linux/route/route_linux.go]: FWIW we don't even need this change if we only bump to the commit which includes the fix: https://github.com/vishvananda/netlink/commit/9de6d08565b3a34150e3e42e0528b8e206953ca4

(That said, I'm also fine with bumping to something slightly newer).

File: go.sum [hunk 2]
```diff
@@ -555,7 +526,6 @@ go.etcd.io/etcd v0.5.0-alpha.5.0.20191023171146-3cf2f69b5738 h1:lWF4f9Nypl1ZqSb4
 go.etcd.io/etcd v0.5.0-alpha.5.0.20191023171146-3cf2f69b5738/go.mod h1:dnLIgRNXwCJa5e+c6mIZCrds/GIG4ncV9HhK5PX7jPg=
 go.mongodb.org/mongo-driver v1.1.2 h1:jxcFYjlkl8xaERsgLo+RNquI0epW6zuy/ZRQs6jnrFA=
 go.mongodb.org/mongo-driver v1.1.2/go.mod h1:u7ryQJ+DOzQmeO7zB6MHyr8jkEQvC8vH7qLUO4lqsUM=
-go.opencensus.io v0.21.0 h1:mU6zScU4U1YAFPHEHYk+3JC4SY7JxgkqS10ZOSyksNg=
```
Comment (joestringer) [inline go.sum:558]: It surprises me how much diff is in this file. Did we just fail to `go mod tidy` until recently or something?

Comment (tgraf) [reply go.sum:558]: Maybe I need to use the exact same go version that we used in 1.7 back then?

**Review by joestringer — APPROVED** [review]
I suspect we've been running these versions (or something very close) in the recent 1.8.8, 1.9.5 releases that we put out last week and we haven't heard much recently so I don't really anticipate any sort of regression from this. Not sure there's anything practical we can do to mitigate the potential risk here if the intent is to pull in fixes that are only available in much newer trees.

File: go.sum (hunk 2 above)
Comment (joestringer) [reply go.sum:558]: That's the intent, yeah.

Comment (joestringer) [reply go.sum:558]: (of course, that means running dumber tooling in practice)

**Comment by tgraf** [comment]
test-backport-1.7

**Comment by brb** [comment]
@tgraf @joestringer Unfortunately, this PR is not enough. We need to backport https://github.com/cilium/cilium/pull/15225/ too. From the provided stack trace we can see that the Go's `net` from the stdlib was doing netlink request.

(Lots of issues, unnecessary work and confusion could have been avoided, if we agreed to backport https://github.com/cilium/cilium/pull/13112 to v1.{7,8,9}).

**Pull request merged**